cleverpush/src/main/java/com/cleverpush/stories/StoryViewListAdapter.java

```diff
@@ -0,0 +1,194 @@
+import com.cleverpush.stories.models.Story;
+
+import java.io.InputStream;
+import java.net.URL;
+import java.util.ArrayList;
+
+import static java.security.AccessController.getContext;
+
+
+public class StoryViewListAdapter extends RecyclerView.Adapter<StoryViewListAdapter.ItemViewHolder> {
+
+    private Context mContext;
+    private ArrayList<Story> stories;
+    private OnItemClicked onItemClicked;
+    TypedArray attrArray;
+    private int DEFAULT_BORDER_COLOR = Color.BLACK;
+    private int DEFAULT_TEXT_COLOR = Color.BLACK;
+
+
+    public StoryViewListAdapter(Context mContext, ArrayList<Story> stories, TypedArray attrArray) {
+        this.mContext = mContext;
+        this.stories = stories;
+        this.attrArray = attrArray;
+    }
+
+    @Override
+    public ItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        LayoutInflater inflater = LayoutInflater.from(mContext);
+        View itemViewStoryHead = inflater.inflate(R.layout.item_view_story, parent, false);
+        return new ItemViewHolder(itemViewStoryHead);
+    }
+
+    @SuppressLint("ResourceType")
+    @Override
+    public void onBindViewHolder(ItemViewHolder holder, int position) {
+        TextView nameTextView = (TextView) holder.itemView.findViewById(R.id.tvTitle);
+        ImageView image = (ImageView) holder.itemView.findViewById(R.id.ivChallenge);
+        nameTextView.setText(stories.get(position).getTitle());
+        nameTextView.setTextColor(attrArray.getColor(R.styleable.StoryView_text_color, DEFAULT_TEXT_COLOR));
+        applyFont(nameTextView, attrArray);
+
+        new Thread(() -> {
+            try {
+                InputStream in = new URL(stories.get(position).getContent().getPreview().getPosterPortraitSrc()).openStream();
+                Bitmap bitmap = BitmapFactory.decodeStream(in);
+                if (bitmap != null) {
+                    image.setImageBitmap(getRoundedCroppedBitmap(bitmap, bitmap.getWidth()));
+                }
+            } catch (Exception ignored) {
+
+            }
+        }).start();
+
+        if (stories.get(position).isOpened()) {
+            image.setBackground(null);
+        } else {
+            GradientDrawable border = new GradientDrawable();
+            border.setShape(GradientDrawable.OVAL);
+            border.setCornerRadii(new float[]{0, 0, 0, 0, 0, 0, 0, 0});
+            border.setColor(0xFFFFFFFF); //white background
+            border.setStroke(5, attrArray.getColor(R.styleable.StoryView_border_color, DEFAULT_BORDER_COLOR)); //black border with full opacity
+            if (Build.VERSION.SDK_INT < Build.VERSION_CODES.JELLY_BEAN) {
+                image.setBackgroundDrawable(border);
+            } else {
+                image.setBackground(border);
+            }
+        }
+
+        image.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (onItemClicked != null) {
```

Comment:
turn into onItemClicked == null then return, because early exits > nested constructs for development in the future